colorbleed/maya/lib.py

```diff
@@ -1292,3 +1312,68 @@ def _get_parent(node):
         _id = get_id(similar_node)
         if _id:
             return _id
+
+
+def set_project_fps():
+    """Set FPS from project configuration
+
+    Returns:
+        None
+
+    """
+
+    int_fps = [15, 24, 5, 30, 48, 50, 60, 44100, 48000]
+    float_fps = [23.976, 29.97, 29.97, 47.952, 59.94]
+
+    fps = lib.get_project_fps()
+
+    if not isinstance(fps, (int, float)):
+        raise ValueError("Set value for project's FPS is not a number. "
+                         "Only accepts floats and integers")
+
+    if int(fps) in int_fps:
```

Comment:
Doesn't it make more sense to first check whether the number is a float and whether it's in the float list? (Because that's the most exact entry point) -> If not found, check if the integer version is a good match.

For example:
```python
if isinstance(fps, float) and fps in float_fps:
    unit = {:d}fps".format(int(fps))

elif int(fps) in int_fps:
    unit = "{:f}fps".format(fps)

else:
    raise ("Unsupported FPS value: `%s`" % fps)

log.info("Updating FPS to '{}'".format(unit))
cmds.currentUnit(time=unit)
```

Note that this is not a bug in the code, it should be functional because none of the float numbers when using `int(number)`  coincide with anything in the int list. Otherwise if `23`, `29`, `47` or `59` would be in the float list it would never match any of those in the float list, because it would always match the one from the int list.

Reply:
Thank you for the pointer

Reply:
Note that my code block isn't entirely correct. The `unit = xx` lines are swapped by accident. :)